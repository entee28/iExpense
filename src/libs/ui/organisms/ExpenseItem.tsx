import React from 'react'
import { formatNumber } from 'libs/utils'
import { Box, Pressable, Text } from 'libs/ui'
import colors from 'libs/ui/colors'
import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome'
import { faArrowRightArrowLeft } from '@fortawesome/free-solid-svg-icons'

type Props = {
  category: string
  amount: number
  icon: string
  currency: string
  onPress: (entry?: Entry) => void
  type: EntryType
  count?: number
  testID?: string
}

export const ExpenseItem = ({
  category,
  amount,
  icon,
  currency,
  type,
  count,
  testID,
  onPress
}: Props) => {
  return (
    <Pressable
      testID={testID}
      onPress={() => onPress()}
      flexDirection="row"
      alignItems="center"
      marginTop={4}>
      {type === 'transfer' ? (
        <Box
          testID="expense_item.transfer_icon"
          width={32}
          height={32}
          alignItems="center"
          justifyContent="center"
          borderRadius={99}
          marginRight={8}
          backgroundColor={colors.primary80}>
          <FontAwesomeIcon color={colors.white} icon={faArrowRightArrowLeft} />
        </Box>
      ) : (
        <Text
          testID="expense_item.category_icon"
          fontSize={32}
          accessibilityLabel="icon">
          {icon}
        </Text>
      )}

      <Box
        flexDirection="row"
        alignItems="center"
        flex={1}
        justifyContent="space-between"
        marginLeft={20}
        borderBottomColor={colors.mono10}
        borderBottomWidth={1}
        paddingBottom={8}
        paddingTop={14}>
        <Box flexDirection="row">
          <Text
            testID="expense_item.category_name"
            fontSize={18}
            accessibilityLabel="category"
            lineHeight={32}
            bold>
            {category}
          </Text>
          {count && (
            <Text
              color={colors.mono40}
              fontSize={18}
              lineHeight={32}
              marginLeft={8}>
              {`x${count}`}
            </Text>
          )}
        </Box>

        <Text
          testID="expense_item.amount"
          color={type === 'income' ? colors.positive100 : colors.mono100}
          fontSize={16}
          lineHeight={32}
          accessibilityLabel="amount"
          bold>
          {formatNumber(amount, {
            currency,
            showCurrency: true,
            decimalCount: 1
          })}
        </Text>
      </Box>
    </Pressable>
  )
}
