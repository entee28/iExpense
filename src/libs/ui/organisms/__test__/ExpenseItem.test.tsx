import { render } from '@testing-library/react-native'
import { ExpenseItem } from '../ExpenseItem'

const category = 'Eating out'
const icon = '🍽️'
const amount = 49000
const currency = '₫'

const item = render(
  <ExpenseItem
    type="expense"
    onPress={() => {}}
    category={category}
    icon={icon}
    amount={amount}
    currency={currency}
  />
)

it('should display information correctly', () => {
  expect(item.getByLabelText('category').children[0]).toEqual(category)
  expect(item.getByLabelText('icon').children[0]).toEqual(icon)
  expect(item.getByLabelText('amount').children[0]).toEqual('49,000.0 ₫')
})

it('should match snapshot', () => {
  expect(item.toJSON()).toMatchSnapshot()
})
