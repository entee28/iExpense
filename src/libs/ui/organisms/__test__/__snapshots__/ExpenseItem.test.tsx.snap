// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <Text
    accessibilityLabel="icon"
    style={
      Array [
        Object {
          "color": "#1A1A1A",
          "fontFamily": "Nunito-Regular",
          "fontSize": 42,
          "lineHeight": undefined,
          "textAlign": "auto",
        },
        Object {},
      ]
    }
  >
    🍽️
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "#1A1A1A1A",
        "borderBottomWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 20,
        "paddingBottom": 8,
        "paddingTop": 14,
      }
    }
  >
    <Text
      accessibilityLabel="category"
      style={
        Array [
          Object {
            "color": "#1A1A1A",
            "fontFamily": "Nunito-Bold",
            "fontSize": 18,
            "lineHeight": 32,
            "textAlign": "auto",
          },
          Object {},
        ]
      }
    >
      Eating out
    </Text>
    <Text
      accessibilityLabel="amount"
      style={
        Array [
          Object {
            "color": "#1A1A1A",
            "fontFamily": "Nunito-Bold",
            "fontSize": 16,
            "lineHeight": 32,
            "textAlign": "auto",
          },
          Object {},
        ]
      }
    >
      49,000.00 ₫
    </Text>
  </View>
</View>
`;
